### PR TITLE
Create footer common to all pages

### PR DESCRIPTION
**Pull request by sharaths21312**
Part of #2, applies a common footer, splitting into multiple PRs to be clearer and more manageable, uses a template present in footerinclude.js

@mrigankpawagi @NaveenMaurya749 do check the content and styling

### CODE REVIEW

**Comment by NaveenMaurya749** [comment]
A few points:
- The footer text would appear better if it were centered when viewport width is less than `600px`. (As it was originally)
- I guess having the footer links to be abruptly underlined when hovered over is visually less elegant than the prior treatment (no underline, color change to `#a2a2ff` when hovered, and `0.3 sec` transition), but it does work in favour of accessibility. Though the beforementioned treatment was already immune to color-blindness. @sharaths21312, what was your original intention here?
- Check if the `../js/footerinclude.js` is properly linked to the "About" page.

Else appears fine on my end, I think.

**Comment by sharaths21312** [comment]
Fixed the first issue
As for the second there's no particular reason for it, it was just what I happened to write at that point and thought it looked fine, it can be changed if needed
The include works fine

**Comment by NaveenMaurya749** [comment]
Alright, then @mrigankpawagi, can you give us some consensus?
Which is better?
This...?
```
.link {
    transition: 0.3s;
    text-decoration: none;
}
.link:hover {
    color: #a2a2ff;
}
```
image
Or this...?
```
link {
    text-decoration: none;
}
.link:hover {
    text-decoration: underline;
}
```
image
(Just the footer link styling)

And after this is addressed, we can probably close off PR 10 and tread towards more important territories, is that right?

File: js/footerinclude.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    this.innerHTML = footerTemplate.innerHTML;
+  }
+}
+
+customElements.define('footer-component', Footer);
```
Comment (mrigankpawagi) [inline js/footerinclude.js:40]: is `customElements` a predefined object in the namespace? This approach is very neat though.

**Review by mrigankpawagi — COMMENTED** [review]
LGTM. I will preview this and let you know. Also apologies for the delayed response.

**Comment by mrigankpawagi** [comment]
> Which is better?

I think the former is better. Underlining looks kind of cluttered. A blue color is neat.

**Comment by mrigankpawagi** [comment]
Also, can we please work with branches instead of forks henceforth? That makes it easier to review by directly checking out.

File: js/footerinclude.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    <br />
+  </h4>
+  <h5>
+    <a class="footer-link" href="https://iisc.ac.in">
+      Indian Institute of Science, Bengaluru - 560012, INDIA.
+    </a>
+  </h5>
+  <br>
+  <h6>
+    Copyright © <span class="copyright">2023</span> by The Department of Computer Science and Automation, Indian
```
Comment (mrigankpawagi) [inline js/footerinclude.js:20]: We can remove CSA & IISc from the Copyright text as this is a repetition from above.

File: js/footerinclude.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  <h4>
+    <a class="footer-link" href="https://csa.iisc.ac.in">
+      The Department of Computer Science and Automation (CSA)
+    </a>
+    <br />
+  </h4>
+  <h5>
+    <a class="footer-link" href="https://iisc.ac.in">
+      Indian Institute of Science, Bengaluru - 560012, INDIA.
+    </a>
```
Comment (mrigankpawagi) [inline js/footerinclude.js:16]: As suggested, color change instead of underline on hover would probably look better.

File: js/footerinclude.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const footerTemplate = document.createElement('template');
+
+footerTemplate.innerHTML = `
+<div class="footer-main">
+<div class="footer-text" id="info">
+  <h1>Databased</h1>
+  <h4>
+    <a class="footer-link" href="https://csa.iisc.ac.in">
+      The Department of Computer Science and Automation (CSA)
```
Comment (mrigankpawagi) [inline js/footerinclude.js:9]: Remove "The" from CSA's name

**Review by mrigankpawagi — CHANGES_REQUESTED** [review]
Feel free to merge without another review once you make these changes.

File: js/footerinclude.js (hunk 1 above)
Comment (sharaths21312) [reply js/footerinclude.js:40]: https://developer.mozilla.org/en-US/docs/Web/API/Window/customElements
It's a part of the language, use to define custom tags which can then be turned into what we want in JS

**Comment by sharaths21312** [comment]
> Also, can we please work with branches instead of forks henceforth? That makes it easier to review by directly checking out.
> 
> Edit: Turns out if not git, the gh CLI makes this very easy with forks too.
@mrigankpawagi the vscode github pull requests extension also allows you to directly checkout a pull request from the GUI too, as well as see the specific changes to files (and comments made)

**Pull request merged**